theano/sandbox/cuda/dnn.py

```diff
@@ -355,88 +374,105 @@ def __init__(self, workmem=None):
         if workmem is None:
             workmem = config.dnn.conv.workmem
         self.workmem = workmem
+        self.inplace = inplace
+        if self.inplace:
+            self.destroy_map = {0: [2]}
         assert self.workmem in ['none', 'small', 'large']
 
     def __setstate__(self, d):
         self.__dict__.update(d)
         if not hasattr(self, 'workmem'):
-            self.workmem = 'small'
+            self.workmem = 'none'
```

Comment:
Because this way, it has the same behaviour as before, and we can avoid testing for the cases that fail there.